qutip/solver/result.py

```diff
@@ -829,14 +829,16 @@ def __add__(self, other):
             raise ValueError("Shared `times` are is required to merge results")
         new = self.__class__(self._raw_ops, self.options,
                              solver=self.solver, stats=self.stats)
+        new.e_ops = self.e_ops
```

Comment:
As far as I can tell, the `e_ops` field was previously never set, because it doesn't get initialized in the constructor and only in `_add_first_traj` ([here](https://github.com/pmenczel/qutip/blob/57cca54994d2b70c9c99761b4726d1676c430b9e/qutip/solver/result.py#L506)). (I noticed it because if I write `result_sum = result1 + result2`, then `result_sum.__repr__()` fails because there is no `e_ops` attribute.)

I don't know why `e_ops` is set in `_add_first_traj` instead of in the constructor, so I left that unchanged.